certificateutil/filter_test.go

```diff
@@ -95,11 +95,10 @@ func TestFilterValidCertificateInfos(t *testing.T) {
 		},
 		{
 			name:             "2 valid, duplicated certs",
-			certificateInfos: []CertificateInfoModel{latestValidCertInfo, earlierValidCertInfo, invalidCertInfo},
+			certificateInfos: []CertificateInfoModel{latestValidCertInfo, latestValidCertInfo},
 			want: ValidCertificateInfo{
 				ValidCertificates:      []CertificateInfoModel{latestValidCertInfo},
-				InvalidCertificates:    []CertificateInfoModel{invalidCertInfo},
```

Comment:
I do not see it from this PR but what happened to the invalid cert section here?

Reply:
This change started with the change above: https://github.com/bitrise-io/go-xcode/pull/247/files/118e1afb0d14f6d240219eeb6239184589bf76fc#diff-f623386f7f310c8d6ced0916bc0daf5dee2991b85dec4cea09d235957fd4fb12R63

After
```
earlierValidCert, privateKey, err := GenerateTestCertificate(serial, teamID, teamName, commonName, earlierValidExpiry)
```
we created a certificate info like this:
```
earlierValidCertInfo := NewCertificateInfo(*latestValidCert, earlierValidCert)
```

And some linter reported an ineffectual assignment to `privateKey`, because it wasn't used.

While in every other cert info, we used the recently created `xyzCert, privateKey`, like:
```
	invalidCert, privateKey, err := GenerateTestCertificate(serial, teamID, teamName, commonName, invalidExpiry)
	if err != nil {
		t.Errorf("init: failed to generate certificate, error: %s", err)
	}
	invalidCertInfo := NewCertificateInfo(*invalidCert, privateKey)
	t.Logf("Test certificate generated: %s", invalidCertInfo)
```

it seems it was just a typo with the `earlierValidCertInfo`.

When I fixed this lint issue, the test case started failing, so I fixed. Based on the test case title we check here if duplicates are detected.

Reply:
As I expected, when the original test case was restored here: https://github.com/bitrise-io/go-xcode/pull/247/commits/bf7ca8d4bfbfbe5064c3664f5ef592ec402cbef7
The test is failing: `    --- FAIL: TestFilterValidCertificateInfos/2_valid,_duplicated_certs (0.00s)`

Reply:
And here the certificate filtering logic is fixed: https://github.com/bitrise-io/go-xcode/pull/247/commits/e1f107859f8c07144b77c407206d72c6e3ebf93a